pydicom/charset.py

```diff
@@ -95,6 +96,167 @@
+            raise e
+        if len(b) != 1 or ((ord(b) & 0x80) ^ msb) != 0:
+            character_set = 'ISO IR 14' if msb == 0 else 'ISO IR 13'
+            msg = 'Given character is out of {}'.format(character_set)
+            raise UnicodeEncodeError('shift_jis', value, i, len(value), msg)
+        encoded += b
+
+    return encoded
+
+
+def _encode_to_jis_x_0208(value, errors='strict'):
+    """Convert a unicode string into JIX X 0208 byte string using iso2022_jp
+    encodings.
+    The escape sequence which is located at the end of the encoded value have
+    to vary depends on the value 1 of SpecificCharacterSet. So we have to
+    trim it and append correct escape sequence manually.
+
+    Parameters
+    ----------
+    value : text type
+        The unicode string as presented to the user.
+    errors : str
+        The behavior of a character which could not be encoded. This value
+        is passed to errors argument of encode funcdtion of str.
+
+    Returns
+    -------
+    byte string
+        The encoded string. If some characters in value could not be encoded to
+        JIS X 0208, it depends on the behavior of iso2022_jp encoder.
+
+    Raises
+    ------
+    UnicodeEncodeError
+        If errors is set to 'strict' and `value` could not be encoded with
+        JIX X 0208.
+    """
+
+    # If errors is not strict, this function is used in fallback.
+    # So keep the tail escape sequence of encoded for backward compatibility.
+    if errors != 'strict':
+        return value.encode('iso2022_jp', errors=errors)
+
+    Encoder = codecs.getincrementalencoder('iso2022-jp')
+    encoder = Encoder()
+
+    encoded = encoder.encode(value[0])
+    if encoded[:3] != ENCODINGS_TO_CODES['iso2022_jp']:
+        raise UnicodeEncodeError(
+            'iso2022_jp', value, 0, len(value),
+            'Given character is out of ISO IR 87')
+
+    for i, c in enumerate(value[1:], 1):
+        try:
+            b = encoder.encode(c)
+        except UnicodeEncodeError as e:
+            e.start = i
+            e.end = len(value)
+            raise e
+        if b[:3] == ENCODINGS_TO_CODES['iso8859']:
+            raise UnicodeEncodeError(
+                'iso2022_jp', value, i, len(value),
+                'Given character is out of ISO IR 87')
+        encoded += b
+    return encoded
+
+
+def _get_escape_sequence_for_encoding(encoding, encoded=None):
+    """Return a escape sequence corresponding to given encoding.  If encoding
+    is shift_jis, return ESC)I or ESC(J depends on thefirst byte of given
+    encoeed.
+
+    Parameters
+    ----------
+    encoding : str
+        An encoding is used to specify  an escape sequence.
+
+    encoded : bytes or str
+        The encoded value is used to choose escase sequence if encoding is
+        shift_jis
+
+    Returns
+    -------
+    string
+        Escape sequence for encoded value.
+    """
+
+    if encoding == 'shift_jis':
+        if encoded is None:
+            return ESC + b'(J'
```

Comment:
Maybe you can make constants from these escape sequences (with the corresponding encoding in the name) for better readability.

Reply:
I made constants whose name are ESC_ISO_IR_14 and ESC_ISO_IR_13 and assigned corresponding escape sequences to them.
https://github.com/pydicom/pydicom/blob/cedde3962e03aa174ea1d2d5428c1eebd7063544/pydicom/charset.py#L246